### PR TITLE
[logger] scrub everything

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

This PR wraps the logger such that anything going to the logger will be scrubbed for credentials before hitting the logger. Log lines should be scrubbed no matter how many receivers are configured in `seelog` (files, console, syslog, etc).

### Motivation

Reduce likelihood of credentials leakage

### Additional Notes

Some performance penalty due to string allocation and regex processing. We're not particularly aggressive in our logging so it shouldn't be too bad, but keep an eye out.

### CODE REVIEW

**Review by arbll — COMMENTED** [review]
This looks good to me. That said I am a bit worried about the potential performance issues we might have in high log throughput scenarios. While it's true that we are not too agressive with the logging I experienced one case where the log where spammed by errors on staging (one per metric that a service was sending). I would sleep better with a minimal `testing.B` benchmark.

**Comment by truthbk** [comment]
Benchmark results:
```
•100% ➜ go test github.com/DataDog/datadog-agent/pkg/util/log -bench=Log -benchtime=10s
goos: darwin
goarch: amd64
pkg: github.com/DataDog/datadog-agent/pkg/util/log
BenchmarkLogVanilla-4                    3000000              4627 ns/op
BenchmarkLogVanillaLevels-4             200000000               88.3 ns/op
BenchmarkLogScrubbing-4                   300000             42679 ns/op
BenchmarkLogScrubbingLevels-4           200000000               66.7 ns/op
BenchmarkLogScrubbingMulti-4              300000             47060 ns/op
PASS
ok      github.com/DataDog/datadog-agent/pkg/util/log   94.461s
```

File: releasenotes/notes/scrub-log-output-b66be1ad3cf26313.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+features:
```
Comment (olivielpeau) [inline releasenotes/notes/scrub-log-output-b66be1ad3cf26313.yaml]: To address before merging: I think we should list this under `other`: release notes reflect user-facing changes, and this PR shouldn't change much to the user-facing behavior of the agent. In the same spirit, we could reword the message below to something like `should prevent leakage of credentials in logs from 3rd-party code or code added in the future`

**Review by olivielpeau — DISMISSED** [review]
Haven't reviewed the code closely, but the approach LGTM.

Let's also test if this has a measurable impact on the CPU usage of a typical Agent logging with a level of `info` and `debug`.

(added a nit on the release note, feel free to address at the last minute)

**Review by arbll — DISMISSED** [review]
LGTM, let's monitor this with nightlies to check it does not bring any huge performance impacts

**Review by arbll — DISMISSED** [review]
🚀

**Comment by truthbk** [comment]
Merging!

**Pull request merged**